2.1-Segwit-Version-1.ipynb

```diff
@@ -0,0 +1,564 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import hashlib\n",
+    "from io import BytesIO\n",
+    "import random\n",
+    "\n",
+    "import util\n",
+    "from test_framework.address import program_to_witness\n",
+    "from test_framework.key import ECKey, ECPubKey, generate_schnorr_nonce\n",
+    "from test_framework.messages import CTransaction, COutPoint, CTxIn, CTxOut, CScriptWitness, CTxInWitness\n",
+    "from test_framework.musig import generate_musig_key, aggregate_schnorr_nonces, sign_musig, aggregate_musig_signatures\n",
+    "from test_framework.script import CScript, OP_1, TaprootSignatureHash\n",
+    "import test_framework.segwit_addr as segwit_addr"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2.1 Taproot Outputs\n",
+    "\n",
+    "* Part 1: Generating a Segwit v1 Output.\n",
+    "* Part 2: Sending funds from the Bitcoin Core wallet.\n",
+    "* Part 3: Constructing a transaction to spend the Segwit v1 output.\n",
+    "* Part 4: Programming Exercise - Spending multiple Segwit v1 output types."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Part 1. Generating a Segwit v1 Output\n",
+    "\n",
+    "Taproot and Schnorr will be made available in Segwit v1 outputs upon activation. Segwit v1 follows the same output script pattern as Segwit V0.\n",
+    "\n",
+    "* Segwit Output: **`[1B Version]` `[Segwit program]`**\n",
+    "* Segwit v1 Output: **`[01]` `[33-Byte Public Key]`**\n",
+    "\n",
+    "The 33-Byte Public key encoding is similar to that of legacy compressed pubkeys, but with a different oddness byte.\n",
+    "* Y-coordinate: even - **`[00]`** or odd - **`[01]`**\n",
+    "* X-coordinate: **`[32B x-coordinate]`**\n",
+    "\n",
+    "Please note:\n",
```

Comment:
I'm going to remove this disclaimer from here. We'll have a disclaimer at the start of the workshop and don't need to repeat that in all the notebooks (although you should mention it when presenting the material).